projects/igniteui-angular/src/lib/tabs/tabs/tab-header.component.ts

```diff
@@ -0,0 +1,121 @@
+import { KEYS } from '../../core/utils';
+
+@Component({
+    selector: 'igx-tab-header',
+    templateUrl: 'tab-header.component.html',
+    providers: [{ provide: IgxTabHeaderBase, useExisting: IgxTabHeaderComponent }]
+})
+export class IgxTabHeaderComponent extends IgxTabHeaderDirective implements AfterViewInit, OnDestroy {
+
+    /** @hidden */
+    @HostBinding('class.igx-tabs__header-item--selected')
+    public get provideCssClassSelected(): boolean {
+        return this.tab.selected;
+    }
+
+    /** @hidden */
+    @HostBinding('class.igx-tabs__header-item--disabled')
+    public get provideCssClassDisabled(): boolean {
+        return this.tab.disabled;
+    }
+
+    /** @hidden */
+    @HostBinding('class.igx-tabs__header-item')
+    public cssClass = true;
+
+    private _resizeObserver: ResizeObserver;
+
+    /** @hidden */
+    constructor(protected tabs: IgxTabsComponent, tab: IgxTabItemDirective, elementRef: ElementRef, private ngZone: NgZone) {
+        super(tabs, tab, elementRef);
+    }
+
+     /** @hidden */
+    @HostListener('keydown', ['$event'])
+    public keyDown(event: KeyboardEvent) {
+        let unsupportedKey = false;
+        const itemsArray = this.tabs.items.toArray();
+        const previousIndex = itemsArray.indexOf(this.tab);
+        let newIndex = previousIndex;
+        const hasDisabledItems = itemsArray.some((item) => item.disabled);
+        switch (event.key) {
+            case KEYS.RIGHT_ARROW:
+            case KEYS.RIGHT_ARROW_IE:
+                newIndex = newIndex === itemsArray.length - 1 ? 0 : newIndex + 1;
+                while (hasDisabledItems && itemsArray[newIndex].disabled && newIndex < itemsArray.length) {
+                    newIndex = newIndex === itemsArray.length - 1 ? 0 : newIndex + 1;
+                }
+                break;
+            case KEYS.LEFT_ARROW:
+            case KEYS.LEFT_ARROW_IE:
+                newIndex = newIndex === 0 ? itemsArray.length - 1 : newIndex - 1;
+                while (hasDisabledItems && itemsArray[newIndex].disabled && newIndex >= 0) {
+                    newIndex = newIndex === 0 ? itemsArray.length - 1 : newIndex - 1;
+                }
+                break;
+            case KEYS.HOME:
+                event.preventDefault();
+                newIndex = 0;
+                while (itemsArray[newIndex].disabled && newIndex < itemsArray.length) {
+                    newIndex = newIndex === itemsArray.length - 1 ? 0 : newIndex + 1;
+                }
+                break;
+            case KEYS.END:
+                event.preventDefault();
+                newIndex = itemsArray.length - 1;
+                while (hasDisabledItems && itemsArray[newIndex].disabled && newIndex > 0) {
+                    newIndex = newIndex === 0 ? itemsArray.length - 1 : newIndex - 1;
+                }
+                break;
+            case KEYS.ENTER:
+                if (!this.tab.panelComponent) {
+                    this.nativeElement.click();

```

Comment:
👀 
Is that needed just to trigger tab selection?
If so, do it through a method or an input. No need to potentially trigger users click handlers by our internal logic

Reply:
This is needed for the routing scenario. In order to trigger the routing we need to call `click()`. If you have a standard `<button>` and press enter it also calls its click handler.

Reply:
Yeah, but this is not a standard button 😄
Anyway, if it is needed for routing I guess we don't have much of a choice.